data-exporter/src/main/java/org/reactome/release/dataexport/UniProtReactomeEntry.java

```diff
@@ -127,104 +160,140 @@ private UniProtReactomeEntry(long dbId, String accession, String displayName) {
 
 			// Add RLEs containing UniProt accession
-			uniprotAccessionToReactomeEvent
+			uniprotReactomeEntryToReactomeEvent
 				.computeIfAbsent(uniprotAccession, k -> new HashSet<>())
 				.addAll(reactionLikeEventIds.stream().map(eventCache::get).collect(Collectors.toSet()));
 
 			for (long reactionLikeEventId : reactionLikeEventIds) {
 				Set<Long> pathwayIds = rleToPathwayId.computeIfAbsent(reactionLikeEventId, k -> new HashSet<>());
-				uniprotAccessionToReactomeEvent
+				uniprotReactomeEntryToReactomeEvent
 					.computeIfAbsent(uniprotAccession, k-> new HashSet<>())
 					.addAll(pathwayIds.stream().map(eventCache::get).collect(Collectors.toSet()));
 			}
 
 			if (count.getAndIncrement() % 10000 == 0) {
 				logger.info("Processed events for " + count.get() + " out of " +
-					uniprotAccessionToReactionLikeEventId.size() + " (" +
+					uniprotReactomeEntryToReactionLikeEventId.size() + " (" +
 					String.format(
 						"%.2f",
-						(100 * (double) count.get()) / uniprotAccessionToReactionLikeEventId.size()
+						(100 * (double) count.get()) / uniprotReactomeEntryToReactionLikeEventId.size()
 					) + "%) " + LocalTime.now()
 				);
 			}
 		});
-		uniprotAccessionToReactomeEventCache.put(graphDBSession, uniprotAccessionToReactomeEvent);
+		uniprotReactomeEntryToReactomeEventCache.put(graphDBSession, uniprotReactomeEntryToReactomeEvent);
 
 		logger.info("Finished computing UniProt to Reactome events");
 
-		return uniprotAccessionToReactomeEvent;
+		return uniprotReactomeEntryToReactomeEvent;
 	}
 
 	/**
-	 * Retrieves, from the graph database, a Map of UniProt accession to the set of identifiers for Reaction Like Events
-	 * in which each UniProt accession participates
+	 * Retrieves, from the graph database, a Map of UniProtReactomeEntry objects to the set of identifiers for
+	 * Reaction Like Events in which each UniProt accession participates
 	 * @param graphDBSession Neo4J Driver Session object for querying the graph database
-	 * @return Map of UniProt accession to set of database identifiers for Reaction Like Events in Reactome
+	 * @return Map of UniProtReactomeEntry objects to set of database identifiers for Reaction Like Events in Reactome
 	 */
-	private static Map<String, Set<Long>> fetchUniProtAccessionToRLEId(Session graphDBSession) {
-		if (uniprotAccessionToReactionLikeEventIdCache.containsKey(graphDBSession)) {
-			return uniprotAccessionToReactionLikeEventIdCache.get(graphDBSession);
+	private static Map<UniProtReactomeEntry, Set<Long>> fetchUniProtReactomeEntryToRLEId(Session graphDBSession) {
+		if (uniprotReactomeEntryToReactionLikeEventIdCache.containsKey(graphDBSession)) {
+			return uniprotReactomeEntryToReactionLikeEventIdCache.get(graphDBSession);
 		}
 
 		logger.info("Computing UniProt to RLE id");
 
 		StatementResult statementResult = graphDBSession.run(
 			String.join(System.lineSeparator(),
 				"MATCH (rgp:ReferenceGeneProduct)<-[:referenceEntity|:referenceSequence|:hasModifiedResidue]-" +
```

Comment:
This query seems a bit complex. Should it have a brief comment above it to clarify what it's doing?

Reply:
Sure - I'll make it clearer.